### PR TITLE
Introduce annotations to handle the fields exclusive to only certain kafka topic names

### PR DESCRIPTION
**Pull request by chengjie8**
## What was the problem?
Currently, we try to fetch all field values in `BieMessagePayload` class from `GenericRecord`. Due to some fields that are not commonly shared from all kafka topics we listen to, it causes exceptions when the `GenericRecord` does not have the field value available.

Associated tickets or Slack threads:
- #2346 

## How does this fix it?[^1]
Introduced two new annotations that support the definition of exclusive fields for Kafka topic event names. This approach will reduce the amount of repetitive and tedious code required when transferring data from GenericRecord to our own `BieMessagePayload` objects. Most of the heavy lifting is now handled by the `BieMessageUtils` class.
- `@Ignore` - mark fields that should be ignored when fetching from a GenericRecord.
- `@TargetEvents` - is used to define a field that is exclusive to certain contention classification events. Valid values are derived from the topic names in the ContentionEvent enum.

## How to test this PR
- `source scripts/kafka-service.sh`
- `./gradlew svc-bie-kafka:test --rerun `
- `./gradlew svc-bie-kafka:integrationTest --rerun`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: svc-bie-kafka/src/main/java/gov/va/vro/services/bie/utils/BieMessageUtils.java [hunk 1]
```diff
@@ -1,65 +1,88 @@
...
-          log.warn(
-              "Type mismatch for field '{}'. Expected type: '{}', Actual value: '{}'",
-              fieldName,
-              field.getType().getSimpleName(),
-              value);
+        boolean matchedTopicName =
+            Arrays.stream(annotationValues).anyMatch(v -> v.equals(contentionEvent.getTopicName()));
+
+        if (matchedTopicName) {
+          invokeSetterMethod(payload, field, genericRecord);
```
Comment (agile-josiah) [inline svc-bie-kafka/src/main/java/gov/va/vro/services/bie/utils/BieMessageUtils.java:43]: Is the if statement still needed? It seems like the if else is calling the same method and everything else is inconsequential/for logging.

**Review by agile-josiah — CHANGES_REQUESTED** [review]
Other than the if statement LGTM

File: svc-bie-kafka/src/main/java/gov/va/vro/services/bie/utils/BieMessageUtils.java (hunk 1 above)
Comment (agile-josiah) [reply svc-bie-kafka/src/main/java/gov/va/vro/services/bie/utils/BieMessageUtils.java:43]: Ah that makes sense. LGTM then

**Review by agile-josiah — APPROVED** [review]

**Review by tejans24 — APPROVED** [review]
LGTM

**Pull request merged**